### PR TITLE
Upgrade for Ink 3

### PR DESCRIPTION
**Pull request by kellymears**
This updates the component to work with Ink 3. 

I couldn't get the tests to pass and I'm really not very familiar with typescript.. but the component is working in my project as expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	"author": {
 		"name": "Vadim Demedes",
 		"email": "vdemedes@gmail.com",
-		"url": "github.com/vadimdemedes"
+		"url": "https://github.com/vadimdemedes"
```
Comment (vadimdemedes) [inline package.json:10]: Could you undo this change?

File: src/SelectInput.js [hunk 2]
```diff
@@ -152,16 +152,13 @@ class SelectInput extends PureComponent {
...
-	render() {
-		return (
-			<StdinContext.Consumer>
-				{({stdin, setRawMode}) => (
-					<SelectInput {...this.props} stdin={stdin} setRawMode={setRawMode}/>
-				)}
-			</StdinContext.Consumer>
-		);
-	}
+const SelectInputWithStdin = props => {
```
Comment (vadimdemedes) [inline src/SelectInput.js:155]: You can remove this component and use `useStdin` hook from inside `<SelectInput>` itself.

**Comment by vadimdemedes** [comment]
Thanks for doing this! Tests are failing, because you're using latest AVA, but config is for the old version. Could you update AVA's config to support Babel? Instructions are here -> https://github.com/avajs/babel.

**Comment by vadimdemedes** [comment]
Hey @kellymears, do you think you'll be able to finish this PR? Totally cool if not.

**Comment by kellymears** [comment]
I have some time tomorrow and I’d love to finish it. Sorry for the delay and thanks x infinity for the rad framework!

File: package.json (hunk 1 above)
Comment (kellymears) [reply package.json]: Addressed in #35ef418.

File: src/SelectInput.js (hunk 2 above)
Comment (kellymears) [reply src/SelectInput.js]: Addressed in bc8a16a. I've never used a hook in a class before 😲. Can you get away with this because it's a PureComponent?

**Comment by kellymears** [comment]
There are actually two issues here, from what I can tell:

**1. `xo` fails to run**

```sh
Error: Error while loading rule '@typescript-eslint/await-thenable': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /Users/kellymears/code/projects/cli/bud/ink-select-input/index.d.ts 
...
```

This was brought up in April https://github.com/xojs/xo/issues/450. 

However, it was already fixed by this PR: https://github.com/xojs/xo/pull/452. 

So, I'm not sure what I'm doing wrong. 

**2. `ava` now runs successfully but tests themselves are failing**

I tried to fix the first error (related to the `Indicator` test), but I wasn't getting anywhere and it's really just because I'm not super familiar with Ava. 


Sorry that this isn't in the bag. I didn't realize that upgrading the ava/xo deps was going to turn into such a thing or I would have tread a little more carefully.

**Comment by vadimdemedes** [comment]
No worries, thanks for investing the time into it! I submitted #27, which is based on this PR and fixes the tests and a bunch of other things ;)
